### PR TITLE
ECM-42 - Link Magento invoices to Balance charges

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Model/BalancepayMethod.php [hunk 1]
```diff
@@ -403,14 +410,26 @@ public function capture(InfoInterface $payment, $amount)
...
                 ->setAmount($amount)
                 ->setBalanceVendorId($balanceVendorId)
                 ->process();
-        }
 
+            $charges = $response->getCharges();
+            foreach ($charges as $charge) {
+                $chargeId = $charge->getId();
+                break;
+            }
```
Comment (ynaxonBalance) [inline Model/BalancepayMethod.php]: couldn't it be just  `$charges[0]`?

File: etc/db_schema.xml [hunk 2]
```diff
@@ -27,4 +27,17 @@
             <column name="entity_id"/>
         </constraint>
     </table>
+    <table name="balance_charges" resource="default" engine="innodb" comment="Balance Charges">
+        <column xsi:type="int" name="entity_id" unsigned="false" nullable="false" identity="true" comment="Entity Id"/>
```
Comment (ynaxonBalance) [inline etc/db_schema.xml]: let's use charge_id as primary key

File: etc/db_schema.xml [hunk 3]
```diff
@@ -27,4 +27,17 @@
...
+        <column xsi:type="int" name="charge_id" nullable="false" comment="Charge Id"/>
```
Comment (ynaxonBalance) [inline etc/db_schema.xml]: this should be a unique index + unsigned (if cannot be used as primary key)

File: etc/db_schema.xml [hunk 4]
```diff
@@ -27,4 +27,17 @@
...
+        <column xsi:type="int" name="invoice_id" unsigned="true"  nullable="false" comment="Invoice Id"/>
```
Comment (ynaxonBalance) [inline etc/db_schema.xml:32]: why is this nullable?

Comment (dhiraj-toptal) [reply etc/db_schema.xml:32]: Its nullable="false" so this is ok right?

Comment (ynaxonBalance) [reply etc/db_schema.xml:32]: correct, my bad

File: Model/BalancepayMethod.php [hunk 5]
```diff
@@ -403,14 +412,22 @@ public function capture(InfoInterface $payment, $amount)
...
-        }
 
+            $charges = $response->getCharges();
+            if (is_array($charges) && isset($charges[0])) {
+                $chargeId = $charges[0]->getId();
+            }*/
+            //$invoiceId = $payment->getCreatedInvoice()->getId();
+            //$invoiceModel = $this->invoice->load($invoiceId);
+        }
+        $payment->setTransactionId('1209348756');
```
Comment (ynaxonBalance) [inline Model/BalancepayMethod.php]: what is this?

File: Model/BalancepayMethod.php [hunk 6]
```diff
@@ -403,14 +412,22 @@ public function capture(InfoInterface $payment, $amount)
                 }
             }
 
-            $this->requestFactory
+            /*$response = $this->requestFactory
```
Comment (ynaxonBalance) [inline Model/BalancepayMethod.php]: Please remove commented code

**Review by ynaxonBalance — COMMENTED** [review]
Do we need the `balance_charges` table if we're saving the charge id as invoice's transaction id?

**Review by ynaxonBalance — APPROVED** [review]
Pay attention to the file 1.3.6.tar.xz, guess you commited it by mistake?

**Review by ynaxonBalance — APPROVED** [review]
Looking good

**Pull request merged**